WhiteDentalClinic/WhiteDentalClinic.DataAccess/Repositories/BaseRepositories/BaseRepository.cs

```diff
@@ -0,0 +1,42 @@
+﻿using Microsoft.EntityFrameworkCore;
+using WhiteDentalClinic.DataAccess.Repositories.BaseRepositories;
+
+namespace WhiteDentalClinic.DataAccess.Repositories
+{
+    public class BaseRepository<TEntity> : IBaseRepository<TEntity>
+        where TEntity : class
+    {
+        protected readonly ApiDbTempContext Context;
+        protected readonly DbSet<TEntity> DbSet;
+        protected BaseRepository(ApiDbTempContext context)   
+        {
+            Context = context;
+            DbSet = context.Set<TEntity>();
+        }
+
+        public List<TEntity> GetAll()
+        {
+            return DbSet.ToList();
```

Comment:
Get all should receive a filter list as a parameter so that it's not required to read all the info from a database table every single time a repository is used. the filter list can be used so that only the data you are interested in is read. Might be a bit more complicated to do it in a generic repository as an entity can have different filters (by id, by dentist, by customer, etc.) and another entity might have completely different filters so maybe this is a step for later on.